staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics.go

```diff
@@ -45,8 +45,8 @@ const (
 )
 
 var (
-	// Use buckets ranging from 5 ms to 10 seconds (admission webhooks timeout at 30 seconds by default).
-	latencyBuckets       = []float64{0.005, 0.025, 0.1, 0.5, 2.5, 5.0, 10.0}
```

Comment:
Was the issue the namespace label addition or the extra buckets?

Reply:
we didn't identify, but the additional buckets for sure amplified the cardinality issue even more. mostly the namespace label is the biggest cardinality contributor.

Reply:
Primarily, the namespace label..

Reply:
How many series in total did you observe this added?

Reply:
It was the churn from e2e tests. They basically create namespaces on a per-test basis.

Reply:
We even have a test which creates 100 namespaces and deletes them.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/namespace.go#L246-L251